pyopia/io.py

```diff
@@ -213,16 +213,14 @@ def combine_stats_netcdf_files(path_to_data, prefix='*'):
                                coords='minimal', compat='override') as ds:
         xstats = ds.load()
 
-    # Check if we have image statistics in the last file, if so, load it.
-    # The last file should contain the entire time series of processed images.
+    # Check if we have image statistics, if so, load it.
     try:
-        ds = xarray.open_dataset(sorted_filelist[-1], group='image_stats')
+        with xarray.open_mfdataset(sorted_filelist, combine='nested', concat_dim='index',
+                                   decode_cf=True, parallel=False,
+                                   group='image_stats') as ds:
```

Comment:
This creates a new dimension "index". We could instead just do

`with xarray.open_mfdataset(sorted_filelist, group='image_stats', preprocess=lambda ds: ds.isel(timestamp=-1)) as ds:`